mkosi/kmod.py

```diff
@@ -55,6 +55,39 @@ def filter_kernel_modules(
     return sorted(modules)
 
 
+def filter_firmware(
+    root: Path,
+    firmware: set[Path],
+    *,
+    include: Iterable[str],
+    exclude: Iterable[str],
+) -> list[Path]:
+    firmwared = Path("usr/lib/firmware")
+    with chdir(root):
+        all_firmware = set(firmwared.rglob("*"))
```

Comment:
We should move this inside of the `if include:` block to optimize things if `FirmwareInclude=` is not set.